sec1_intro_and_setup/test_app/app/models/application_record.rb

```diff
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
```

Comment:
Before getting into `application_record.rb`:
>About `app/controllers/application_controller.rb`:
>Controllers are just ruby classes that inherit from a class called `ActionController`. As you add more models to your project you will have more controllers to deal with. `application_controller.rb` is required for any Rails project because it inherits from ActionController::Base and all future controllers intern inherit from it, giving them the functionality of Controllers.
You’ll see a lot of “application_[insert relative title here].rb” files: `application_controller.rb`, `application_helper.rb`, `application_record.rb`. In most cases, these represent a global way of interacting with the app, an intermediary that inherits from a base class and then is inherited by future classes or both.

`application_record.rb` inherits from `ActiveRecord::Base` all subsequent models in your app will inherit from `ApplicationRecord` , similar to how `application_controller` makes the functionality of the `ActionController` available to all other Controllers. The `ApplicationRecord` is an abstract class which means `ApplicationRecord` is not mapped to a database table. The `ApplicationRecord` class allows one to write methods and include code that will be inherited by all of our models.

Reply:
You are implying that without application record class, we are not able to write methods and include code that will be inherited by all models.
That is 100000% incorrect

1. Why must all models inherit from application record? Why can't it just inherit from `ActiveRecord::Base` directly?
2. Please summarise your answer from your own understanding.

Reply:
`ActiveRecord::Base.include(MyFeature)` will forever include the `MyFeature` and any class inheriting from it also includes `MyFeature` even if they don't want it. This also can lead to problems if two engines/gems tried to include the same-named methods (overwriting and the previous one is nowhere to be found).

`ApplicationRecord` create an abstract class and localise the `MyFeature` to only the models which are inheriting from `ApplicationRecord`.

Reply:
ok. So, based on your asnwer, why then have `ApplicationRecord` directly inherit ActiveRecord since it can:
>can lead to problems if two engines/gems tried to include the same-named methods (overwriting and the previous one is nowhere to be found)

Reply:
`ApplicationRecord` acts as the only entry for all customisation and modification for the application and `ActiveRecord` remains in its original form (away from any modifications made).

Reply:
You aren't answering the question but keep going round and round. 
Lets move on.